cruise.umple.nebula/src/cruise/umple/modeling/handlers/ModelingAssociationsGenerationPointsHandler.java

```diff
@@ -1108,6 +1108,8 @@ private static GenMethod registerAddOrMoveAt(GenAssociation association) {
 		addOrMoveAt.setReturnType(CPPTypesConstants.BOOL);
 		addOrMoveAt.addParameter(new GenMethodParameter(normalizedType, newInstance));
 		addOrMoveAt.addParameter(new GenMethodParameter(CPPTypesConstants.INTEGER, "index"));
+		String addOrMoveAtName = association.addOrMoveAtInstance(source.getRole());

```

Comment:
These two lines request the association manager to add a new instance or move an existing one to a specific role and then set the name of the result.

Reply:
Add these comments in the code itself.